src/v2/Components/ArtworkFilter/index.tsx

```diff
@@ -167,29 +171,37 @@ export const BaseArtworkFilter: React.FC<
    * and trigger a reload.
    */
   useDeepCompareEffect(() => {
-    // @ts-expect-error STRICT_NULL_CHECK
-    Object.entries(filterContext.filters).forEach(
+    Object.entries(filterContext.filters ?? {}).forEach(
       ([filterKey, currentFilter]) => {
-        // @ts-expect-error STRICT_NULL_CHECK
-        const previousFilter = previousFilters[filterKey]
+        const previousFilter = previousFilters?.[filterKey]
         const filtersHaveUpdated = !isEqual(currentFilter, previousFilter)
 
         if (filtersHaveUpdated) {
           fetchResults()
 
-          tracking.trackEvent(
-            commercialFilterParamsChanged({
-              changed: JSON.stringify({
-                // @ts-expect-error STRICT_NULL_CHECK
-                [filterKey]: filterContext.filters[filterKey],
-              }),
-              contextOwnerId: contextPageOwnerId,
-              contextOwnerSlug: contextPageOwnerSlug,
-              // @ts-expect-error STRICT_NULL_CHECK
-              contextOwnerType: contextPageOwnerType,
-              current: JSON.stringify(filterContext.filters),
+          if (filterKey === "page") {
+            tracking.trackEvent({
+              action: ActionType.clickedChangePage,
+              context_module: ContextModule.artworkGrid,
+              context_page_owner_type: contextPageOwnerType,
+              context_page_owner_id: contextPageOwnerId,
+              context_page_owner_slug: contextPageOwnerSlug,
+              page_current: previousFilter,
+              page_changed: currentFilter,
```

Comment:
**change requested:** let's import `ClickedChangePage` from Cohesion (the version should be updated automatically, but if you don't see the event, try updating it with `yarn`) and explicitly type this event. It could look like:
```ts
const PageTrackingParams: ClickedChangePage = {
action: ActionType.clickedChangePage,
context_module: ContextModule.artworkGrid,
context_page_owner_type: contextPageOwnerType,
context_page_owner_id: contextPageOwnerId,
context_page_owner_slug: contextPageOwnerSlug,
page_current: previousFilter,
page_changed: currentFilter
}
```

That way we're 100% sure we have all of the desired attributes for this event and they're typed correctly :)

Reply:
Sounds good!